[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35688 [quant] Move quantization tests into test/quantization
* #35571 [quant][graphmode] Quantization support for at::sort
* #35558 [quant][graphmode][refactor] getGeneralTensorInputs(Node*) -> getPassThroughInputs(Value*)
* #35557 [quant][graphmode][fix] docs for InsertObservers
* **#35509 [quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`**

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20742138](https://our.internmc.facebook.com/intern/diff/D20742138)

jerryzh168 committed:
[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on test/test_jit.py:
```diff
@@ -1612,6 +1612,36 @@ def forward(self, x, y):
+            m = convert_script(m, True)
+            FileCheck().check_not("aten::add") \
+                       .check_not("aten::add_") \
+                       .check_not("aten::relu") \
```
Should we also check for `quantized::add` , `quantized::relu`?

supriyar commented on torch/csrc/jit/passes/quantization_patterns.h:
```diff
@@ -219,6 +253,7 @@ graph(%a_quant, %b_scalar, %alpha):
        add_scalar_out,
        quantized_add_scalar_out,
        add_scalar_filter},
+      {"quantized::cat", cat, quantized_cat},
```
is this change related to this PR?

jerryzh168 replied:
this is just moving around quantized cat

supriyar reviewed (approved):
looks good. small comments.

jerryzh168 committed:
Update on "[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20742138](https://our.internmc.facebook.com/intern/diff/D20742138)

[ghstack-poisoned]
